src/xtgeo/xyz/_xyz.py

```diff
@@ -56,23 +55,32 @@ def __init__(self, *args, **kwargs):
         logger.info("XYZ Instance initiated (base class) ID %s", id(self))
 
     @property
-    @abc.abstractmethod
     def xname(self):
         """ Returns or set the name of the X column."""
         return self._xname
 
+    @xname.setter
+    def xname(self, newname):
+        self._name_setter(newname)
+
     @property
-    @abc.abstractmethod
     def yname(self):
         """ Returns or set the name of the Y column."""
         return self._yname
 
+    @yname.setter
+    def yname(self, newname):
+        self._name_setter(newname)
+
     @property
-    @abc.abstractmethod
     def zname(self):
         """ Returns or set the name of the Z column."""
         return self._zname
 
+    @zname.setter
+    def zname(self, newname):
+        self._name_setter(newname)
+
     def _name_setter(self, newname):
```

Comment:
Could we not also remove this generalised method and instead have the setters individually?

We could have the check for wether the name is one of the columns in a different function of course, if we are trying to avoid duplicate code.

Reply:
Some additional thoughts on this. There's a strict difference in how `yname`, `xname` and `zname` are handled, and the `tname` etc. That is not a problem in itself. But I am unsure if it can be confusing?

I'm making a dataframe that has a sane name for `Y_UTMN`, but nothing else.

```
>>> import pandas as pd
>>> data = pd.DataFrame({"Y_UTMN": [1, 2], "SOME": [3, 4]})
>>> test = polygons.Polygons()
>>> test.dataframe = data
>>> test.dataframe
   Y_UTMN  SOME
0       1     3
1       2     4
```

Setting the `yname` will modify the column name, while setting the `tname` will just make `tname` refer to said column given that it exists (otherwise it will fail):

```
>>> test.yname = "NEW_YNAME"
>>> test.dataframe
   NEW_YNAME  SOME
0          1     3
1          2     4
>>> test.name = "SOME"
>>> test.dataframe[test.tname]
0    3
1    4
Name: SOME, dtype: int64
```

The curiosity occurs when we try to do anything with e.g. `xname`

```
>>> test.xname
'X_UTME'
>>> test.xname = "SOMETHING"
>>> test.xname
'SOMETHING'
>>> test.dataframe
   NEW  SOME
0    1     3
1    2     4
```

Now the `yname` will refer to a proper column, while `xname`, although a string, does not exist in the dataframe. Thus looking them up will be quite different:

```
>>> test.dataframe[test.yname]
0    1
1    2
Name: NEW, dtype: int64
>>> test.dataframe[test.xname]
Traceback (most recent call last):
  File "/Users/LPHA/venvs/xtgeo_env/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 2898, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 101, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'SOMETHING'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/LPHA/venvs/xtgeo_env/lib/python3.8/site-packages/pandas/core/frame.py", line 2906, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/Users/LPHA/venvs/xtgeo_env/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 2900, in get_loc
    raise KeyError(key) from err
KeyError: 'SOMETHING'
```

Another interesting behaviour is that if we rename the `yname` to an existing column name:

```
>>> test.yname = "SOME"
>>> test.dataframe
   SOME  SOME
0     1     3
1     2     4
>>> test.dataframe[test.yname]
   SOME  SOME
0     1     3
1     2     4
```

This could end up being difficult to debug for any methods assuming that the result of `test.dataframe[test.yname]` is a `pandas.Series` instead of a `pandas.DataFrame`

Please consider the example above more in a scenario where there's a typo for e.g. `xname`.

Perhaps the fact that pandas will not fail if a column name suggested to be replaced does not exist:

`self._df.rename(columns={oldname: newname}, inplace=True)`

is part of the reason why this goes undiscovered (I don't disagree with pandas though)

-----------------

Suggestion:

If we heavily rely on the existence of `xname`, `yname` and `zname`, we could check for the column names when setting the dataframe, and throw if they don't exist.